clang/tools/libclang/CMakeLists.txt

```
@@ -106,7 +106,7 @@ if (LLVM_EXPORTED_SYMBOL_FILE)
                      DEPENDS ${LIBCLANG_VERSION_SCRIPT_FILE})
 endif()
 
-if((NOT (WIN32 OR CYGWIN) AND LLVM_ENABLE_PIC) OR
+if((NOT CYGWIN AND LLVM_ENABLE_PIC) OR
   ((WIN32 OR CYGWIN) AND NOT LIBCLANG_BUILD_STATIC))
```

I'd just do
```suggestion
if(LLVM_ENABLE_PIC OR ((WIN32 OR CYGWIN) AND NOT LIBCLANG_BUILD_STATIC))
```

since `LIBCLANG_BUILD_STATIC` is documented to build a static libclang *in addition to* the shared libclang

---

done

---

My build just failed with too many exports.  But I bet it can be fixed in a better way so it'd be fine with me to merge this to unbreak actually entirely working platforms and leave me to fix it later.

---

I opened #138351 to try to figure out how to fix this on Cygwin.  Maybe for our static build bootstrapping workaround we need to do both `-DLLVM_ENABLE_PIC=OFF -DLIBCLANG_BUILD_STATIC=ON`.

---

I did confirm that `-DLLVM_ENABLE_PIC=OFF -DLIBCLANG_BUILD_STATIC=ON` works to build on Cygwin with GCC, so we do still have a workaround with these changes.